### DISCUSSION TITLE
Running CapRover locally via Vagrant

### DISCUSSION THREAD

**Discussion by abulka** [caprover/caprover#1751] [Q&A]
Since dokku doesn't support ARM and caprover does, I needed and got caprover running on my Mac Mini M1 using Vagrant. 

This use case is possible and works, and I even installed Wordpress from the dashboard. A pity it is not better documented as I'd imagine it is a common scenario for local development - the existing local install documentation gets me 70% of the way there but I had to figure out quite a few things on my own.  Compare this to the dokku project which has a Vagrant project and documentation as first class citizens. Anyway, I can supply my Vagrantfile and installation steps if anyone is interested.

What I don't understand is why I need to have 
```
  config.vm.network "forwarded_port", guest: 3000, host: 3000
  config.vm.network "forwarded_port", guest: 80, host: 80
  config.vm.network "forwarded_port", guest: 443, host: 443
```
in my `Vagrantfile` in order for the http://captain.caproverinstance.local name to work. Note I have /etc/hosts
```192.168.0.219    captain.caproverinstance.local``` and set the root domain in caprover to `caproverinstance.local` via the dashboard.

If I comment out the vagrant port 80 to 80 mapping, the url fails. The port mapping in the vagrantfile also means my localhost:80 is now taken up by caprover. Trying to get my head around this. Why would port 80, 300 and 443 on localhost be needed to get caprover properly running on my vagrant VM on 192.168.0.219 ?

**Comment by githubsaturn** [comment]
Not sure how Vagrant port mapping works. But the nginx container that CapRover runs uses `host` mode for [port mapping](https://docs.docker.com/engine/reference/commandline/service_create/#:~:text=The%20mode%20to%20use%20for%20binding%20the%20port%2C%20either%20ingress%20or%20host.%20Defaults%20to%20ingress%20to%20use%20the%20routing%20mesh.).



The reason why this has to be done this way is that otherwise the source IP in nginx logs will be masked behind a private IP:

https://serverfault.com/questions/778798/nginx-reverse-proxy-in-docker-container-wrong-ip-logged
